OutOfSchool/OutOfSchool.BusinessLogic/Extensions/SearchResultExtensions.cs

```diff
@@ -4,5 +4,10 @@ namespace OutOfSchool.BusinessLogic.Extensions;
 public static class SearchResultExtensions
 {
     public static bool IsNullOrEntitiesEmpty<T>(this SearchResult<T> searchResult)
-        => searchResult == null || searchResult.Entities.Count == 0;
+        => IsNull(searchResult) || searchResult.Entities is null || searchResult.Entities.Count == 0;
+
+    public static bool IsNullOrTotalAmountIsZero<T>(this SearchResult<T> searchResult)
+        => IsNull(searchResult) || searchResult.TotalAmount == 0;
+
+    private static bool IsNull<T>(SearchResult<T> searchResult) => searchResult == null;
```

Comment:
(Overengineering) Please avoid extracting so simple code into methods.

Reasons:
1. Method that does only 1 simple operation, doesn't decrease cognitive complexity.
2. This specific method is overcomplicated, it could be replaced by simple `bool IsNull(object o) => o is null;` 
- we don't need input argument specific type as we don't use it inside method;
- we don't need this method to be generic (and thus clutter the generated code with many method specifications) as generic type is not used.
3. Due to private nature - this method is never reused outside method, so doesn't help to "improve" whole project code style
4. Last but not least - there is a "rule of 3 duplicates". And here it is violated as well